build: add HIPO submodule and respect its updated build system

c-dilks opened the pull request:
- add HIPO as a submodule
- add convenience script `installHIPO`
- update README and CI

c-dilks commented:
Awaits #76, since Meson needs Python3. Alternative is #75, but IMO it's better to use the CCDB version that is used upstream.

c-dilks commented on .github/workflows/main.yml:
```diff
@@ -49,14 +49,13 @@ jobs:
         with:
           repository: gavalian/hipo
-          ref: 4.0.1 # if you want to use a fixed tag; remove if you want to use `main`
+          ref: 20094876df040ec234dd376fa527cf58442b775a
```
@dglazier would you prefer to use a fixed version of HIPO for CI tests, or use its `master` branch? The `master` branch means you'll always be testing the latest version, whereas using a fixed version means someone has to remember to periodically update the `ref` here.

dglazier replied:
I think with the fixed version. Actually I am not sure why, when we went to an external hipo I did not just make it a submodule. This way the version is fixed in any clas12root tag and should always work. Perhaps it makes sense to do it that way ?

c-dilks replied:
That's fine too. Would you like me to use a submodule in this PR? This would then be more consistent with your other dependencies.

dglazier replied:
Yes I think that would be best, thanks.

c-dilks commented:
I've added an installation script, `installHIPO`, which locally installs the new `hipo` submodule. The submodule path is `hipo_src/`, so that the installation prefix may be `hipo/` (the default choice from `installHIPO`); this hopefully minimizes confusion for users. Read the script, try it out, and let me know what you think.

I've also updated the main `README.md` and CI config.

Also, is the `hipo4` directory supposed to be here still? It is listed in `.gitignore`, and I only see `RunRoot/Ex6_NtupleWriter.C` using those headers; it is also mentioned in the `README.md`, as part of the `ROOT_INCLUDE_PATH` recommendations.

dglazier commented:
Hi Chris,
  I downloaded your branch and everything ran as it should, so it is good for me. 
  The hipo4 directory provides an interface for using hipo files as a ntuple store with variable sized branches. It actually works pretty nicely and is used in chanser (which requires the instructions given) so we should keep it for now.
  If you are happy with everything I will accept this pull request.
Cheers
Derek

c-dilks commented:
> The hipo4 directory provides an interface for using hipo files as a ntuple store with variable sized branches. It actually works pretty nicely and is used in chanser (which requires the instructions given) so we should keep it for now.

Should `hipo4` be removed from `.gitignore` then, so the files are tracked?

> If you are happy with everything I will accept this pull request.

I'm happy, go for it! Thanks for testing.

dglazier commented:
Yes I will remove it from .gitignore. Thanks.
